src/prefect/engine/task_runner.py

```diff
@@ -613,6 +616,34 @@ def load_results(
         """
         return state, upstream_states
 
+    def check_target(self, state: State, inputs: Dict[str, Result]) -> State:
+        """
+        Checks if a Result exists at the task's target.
+
+        Args:
+            - state (State): the current state of this task
+            - inputs (Dict[str, Result]): a dictionary of inputs whose keys correspond
+                to the task's `run()` arguments.
+
+        Returns:
+            - State: the state of the task after running the check
+        """
+        result = self.task.result
```

Comment:
```suggestion
        result = self.result
```
The task runner stores a copy of the result object; this is important in situations where a user provides a result to the _flow_ object but not to individual tasks